examples/fedopt_example/client.py

```diff
@@ -1,176 +1,93 @@
-            epoch_metrics.update_performance(predicted, labels)
-
-    log_str = epoch_metrics.summarize()
-    # Local client logging.
-    log(
-        INFO,
-        f"Client Validation Loss: {loss/n_batches}\nClient Validation Metrics:{log_str}",
-    )
-    return loss / n_batches, epoch_metrics.results
-
-
-class NewsClassifier(NumpyFlClient):
-    def __init__(self, data_path: Path, device: torch.device) -> None:
-        super().__init__(data_path, device)
-        self.parameter_exchanger = FullParameterExchanger()
-
-    def setup_client(self, config: Config) -> None:
-        super().setup_client(config)
+from fl4health.checkpointing.checkpointer import TorchCheckpointer
+from fl4health.clients.basic_client import BasicClient
+from fl4health.utils.losses import LossMeter, LossMeterType
+from fl4health.utils.metrics import MetricMeterManager
+
+
+class NewsClassifierClient(BasicClient):
+    def __init__(
+        self,
+        data_path: Path,
+        device: torch.device,
+        loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
+        checkpointer: Optional[TorchCheckpointer] = None,
+    ) -> None:
+        super(BasicClient, self).__init__(data_path, device)
+        self.checkpointer = checkpointer
+        self.train_loss_meter = LossMeter.get_meter_by_type(loss_meter_type)
+        self.val_loss_meter = LossMeter.get_meter_by_type(loss_meter_type)
+
+        self.model: nn.Module
+        self.optimizer: torch.optim.Optimizer
+
+        self.train_loader: DataLoader
+        self.val_loader: DataLoader
+        self.num_train_samples: int
+        self.num_val_samples: int
+        self.learning_rate: float
+        self.weight_matrix: torch.Tensor
+        self.vocabulary: Vocabulary
+        self.label_encoder: LabelEncoder
+        self.batch_size: int
+
+        # Need to track total_steps across rounds for WANDB reporting
+        self.total_steps: int = 0
+
+    def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
         sequence_length = self.narrow_config_type(config, "sequence_length", int)
-        batch_size = self.narrow_config_type(config, "batch_size", int)
-        vocab_dimension = self.narrow_config_type(config, "vocab_dimension", int)
-        hidden_size = self.narrow_config_type(config, "hidden_size", int)
-        vocabulary = Vocabulary.from_json(self.narrow_config_type(config, "vocabulary", str))
-        label_encoder = LabelEncoder.from_json(self.narrow_config_type(config, "label_encoder", str))
+        self.batch_size = self.narrow_config_type(config, "batch_size", int)
 
-        train_loader, validation_loader, num_examples, weight_matrix = construct_dataloaders(
-            self.data_path, vocabulary, label_encoder, sequence_length, batch_size
+        train_loader, validation_loader, _, weight_matrix = construct_dataloaders(
+            self.data_path, self.vocabulary, self.label_encoder, sequence_length, self.batch_size
```

Comment:
Note: `self.vocabulary` and `self.label_encoder` are initialized in `setup_client` before the call to `super().setup_client() `to ensure their availability

Reply:
Maybe adding a comment for future reference when people are looking at the code?

Reply:
Good call.